src/Button.js

```diff
@@ -27,6 +40,21 @@ Button.defaultProps = {
   theme
 }
 
+const Group = ({children, ...rest}) => {
+  const newChildren = React.Children.map(children, child => React.cloneElement(child, {grouped: true}))
```

Comment:
We could avoid using `React.Children` and `React.cloneElement` by having `Group` apply styles its children. Here's a codesandbox example: https://codesandbox.io/s/buttongroup-u2h46

I don't know that approach is any better 🤷‍♂ What do you think?